packages/react-native/scripts/cocoapods/codegen.rb

```diff
@@ -11,8 +11,8 @@
 # - dir_manager: a class that implements the `Dir` interface. Defaults to `Dir`, the Dependency can be injected for testing purposes.
 # @throws an error if it could not find the codegen folder.
 def build_codegen!(react_native_path, relative_installation_root, dir_manager: Dir)
-    codegen_repo_path = "#{relative_installation_root}/#{react_native_path}/../react-native-codegen";
-    codegen_npm_path = "#{relative_installation_root}/#{react_native_path}/../@react-native/codegen";
+    codegen_repo_path = "#{basePath(react_native_path, relative_installation_root)}/../react-native-codegen";
```

Comment:
Following up with @NickGerleman suggestion, probably it is better to use some system APIs...
```suggestion
    codegen_repo_path = File.join(basePath(react_native_path, relative_installation_root), "..", "react-native-codegen")
```

Reply:
Changing this seems to break some codegen tests, I'll try to investigate what's going on latter today